docs/contributor/arch/017-otel-log-agent-fault-toleration.md

```diff
@@ -0,0 +1,110 @@
+
+![Exporter Batcher Flow](../assets/log-agent-exporter-batcher-flow.svg "Exporter Batcher Flow")
+
+#### Exporter Persistent Sending Queue
+When persistent queue is enabled, the batches are being buffered using the provided storage extension - filestorage is a popular and safe choice. If the collector instance is killed while having some items in the persistent queue, on restart the items will be picked and the exporting is continued.
+
+There are two types of file storage that can back a persistent queue: the node filesystem or a persistent volume (PV).
+
+Using the node filesystem carries some risks. Misconfigurations can lead to disk overflow, potentially crashing the node or even the entire cluster. Additionally, heavy disk I/O can degrade cluster performance—something we previously observed with Fluent Bit. Another limitation is that queue size can currently only be restricted by batch count, not by volume (MB), requiring some estimations. However, the upstream project is actively working on adding size-based limits.
+
+PV-based storage avoids these issues but cannot be used with a DaemonSet.
+
+Overall, we have had positive experiences using node filesystem-based buffering in Fluent Bit, making it a viable solution.
+
+#### Filelog Receiver Offset Tracking
+The Filelog Receiver can persist state information on storage (typically the node’s filesystem), allowing it to recover after crashes. It maintains the following data to ensure continuity:  
+
+- **Number of tracked files** (*knownFiles*).  
+- **For each tracked file:**  
+  - **File fingerprint** (*Fingerprint.first_bytes*) – a unique identifier for the file.  
+  - **Byte offset** (*Offset*) – the position from which the receiver resumes reading.  
+  - **File attributes** (*FileAttributes*) – metadata such as the file name.  
+
+### Agent-to-Backend vs Agent-to-Gateway-to-Backend
+Let's compare the following architectures Agent-to-Backend and Agent-to-Gateway-to-Backend. Each has its own trade-offs.
+
+#### Agent-to-Backend
+Direct communication between the agent and the backend. Gateway is optional and only handles OTel logs.
+
+Pros:
+
+- Reduced latency.
+- Direct communication ensures minimal failure points.
+- If the backend enforces per-connection rate limiting, only high-volume agents get throttled. It can be beneficial since often most workloads in a cluster generate minimal logs, while a few may produce a large volume.
+- A multi-pipeline setup in the agent can be implemented more naturally, with a single [OTel pipeline](https://opentelemetry.io/docs/collector/architecture/#pipelines) per LogPipeline.
+
+Cons:  
+
+- Every gateway enrichment feature must be duplicated in the agent and tested separately.  
+- Credential rotations require a full DaemonSet rollout and restart.
+- Potentially higher costs, as each agent must implement the full telemetry pipeline.  
```

Comment:
do we have any numbers to back that claim?

Reply:
No numbers, just our experience with k8sattributesprocessor. If log enrichment is happening in the agents, we might end up having many more instances of these potentially memory-heavy processors.